activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Add middleware for automatic shard swapping.
+
+    Provides a basic middleware to perform automatic shard swapping. Applications will provide a resolver which will determine for an individual request which shard should be used. Example:
+
+    ```ruby
+    config.active_record.shard_resolver = ->(request) {
+      subdomain = request.subdomain
+      tenant = Tenant.find_by_subdomain!(subdomain)
```

Comment:
Will an `ActiveRecord::NotFoundError` be raised and result in a 404 in the event that the tenant can't be found?

Reply:
In this example, yes, but the resolver design lets apps handle that internally.